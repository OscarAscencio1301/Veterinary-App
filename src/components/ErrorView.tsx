import { ReactElement } from "react"

export const ErrorView = ({ children }: { children: ReactElement }) => {
    return (
        <div aria-label="error" className="bg-red-800 text-white rounded py-2"><p>{children}</p></div>
    )
}
