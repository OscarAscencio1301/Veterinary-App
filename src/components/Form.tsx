

export const Form = () => {
  return (
    <div className="text-center">
      <h2 className="font-black text-3xl">Seguimiento</h2>
      <p className="text-lg my-5">Añade Pacientes y <span className="text-indigo-600 font-bold">Administralos</span></p>
      <form className="bg-white shadow-md rounded-md py-10 px-5">
        <div className="flex flex-col text-left gap-2 py-2">
          <label className="text-gray-700 font-bold"
            htmlFor="mascota">Nombre Mascota</label>
          <input type="text" placeholder="Nombre Mascota" id="mascota" className="p-2 border placeholder-gray-400 rounded-md" />
        </div>
        <div className="flex flex-col text-left gap-2 py-2">
          <label className="text-gray-700 font-bold"
            htmlFor="propietario">Nombre Propietario</label>
          <input type="text" placeholder="Nombre Propietario" id="propietario" className="p-2 border placeholder-gray-400 rounded-md" />
        </div>
        <div className="flex flex-col text-left gap-2 py-2">
          <label className="text-gray-700 font-bold"
            htmlFor="email">Correo Electrónico</label>
          <input type="email" placeholder="Email" id="email" className="p-2 border placeholder-gray-400 rounded-md" />
        </div>
        <div className="flex flex-col text-left gap-2 py-2">
          <label className="text-gray-700 font-bold"
            htmlFor="alta">Alta</label>
          <input type="date" id="alta" className="p-2 border placeholder-gray-400 rounded-md" />
        </div>
        <div className="flex flex-col text-left gap-2 py-2">
          <label className="text-gray-700 font-bold"
            htmlFor="sintomas">Sintomas</label>
          <textarea id="sintomas" className="p-2 border placeholder-gray-400 rounded-md" />
        </div>
        <button type="submit" className="bg-indigo-600 block w-full p-3 rounded-md text-white mt-3 hover:bg-indigo-800 cursor-pointer transition-colors duration-200">Agregar</button>
      </form>
    </div>
  )
}
