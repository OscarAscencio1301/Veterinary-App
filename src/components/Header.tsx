

export const Header = () => {
  return (
    <>
    <h1 className="font-black text-5xl text-center md:w-3/4 mx-auto">Seguimiento Paciente <span className="text-indigo-600">Veterinaria Github</span></h1>
    </>
  )
}
