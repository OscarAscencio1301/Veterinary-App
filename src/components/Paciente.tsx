import { usePacient } from "../hooks/usePacient"
import { Pacient } from "../interfaces/pacients"

export const Paciente = ({date, email, name, pet, sintoms, id}:Pacient) => {
    
    const {deletePacient, activePacient} = usePacient()

    const deleteBTN = () => {
        deletePacient(id || '')
    }

    const updateBTN = () => {
        activePacient({date, email, name, pet, sintoms, id})
    }

    return (
        <div className="my-3 p-10 bg-white py-10 shadow-md rounded-md" aria-label="pacient">
            <p className="font-bold py-3 text-gray-700 uppercase">Mascota: <span className="font-normal normal-case">{pet}</span></p>
            <p className="font-bold py-3 text-gray-700 uppercase">Propietario: <span className="font-normal normal-case">{name}</span></p>
            <p className="font-bold py-3 text-gray-700 uppercase">Correo Electr√≥nico: <span className="font-normal normal-case">{email}</span></p>
            <p className="font-bold py-3 text-gray-700 uppercase">Alta: <span className="font-normal normal-case">{date}</span></p>
            <p className="font-bold py-3 text-gray-700 uppercase">Sintomas: <span className="font-normal normal-case">{sintoms}</span></p>
            <div className="flex justify-between">
            <button type="button" className="py-2 px-10 bg-indigo-600 hover:bg-indigo-900 rounded-md uppercase text-white font-bold" onClick={updateBTN}>Editar</button>
                <button type="button" className="py-2 px-10 bg-red-600 hover:bg-red-900 rounded-md uppercase text-white font-bold" onClick={deleteBTN}>Eliminar</button>
            </div>
        </div>
    )
}
