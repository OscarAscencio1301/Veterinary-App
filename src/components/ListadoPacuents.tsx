import { Paciente } from "./Paciente"

export const ListadoPacientes = () => {
  return (
    <div className="text-center col-span-2 h-screen overflow-y-auto">
      <h2 className="font-black text-3xl">Listado de Pacientes</h2>
      <p className="text-lg my-5">Administra tus <span className="text-indigo-600 font-bold">Pacientes y Citas.</span></p>
      <Paciente />
      <Paciente />
      <Paciente />
    </div>
  )
}
