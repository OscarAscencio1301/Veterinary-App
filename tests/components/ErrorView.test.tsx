import { render, screen } from '@testing-library/react'
import { ErrorView } from '../../src/components/ErrorView';

describe('Testing component ErrorView', () => {
    test('Snapshot', () => {
        const { container } = render(<ErrorView><p>Error Testing</p></ErrorView>)
        expect(container).toMatchSnapshot()
    });
    test('Initial Values', () => {
        render(<ErrorView><p>Error Testing</p></ErrorView>)
        expect(screen.getByText('Error Testing')).toBeTruthy()
    });
    test('ClassList', () => {
        render(<ErrorView><p>Error Testing</p></ErrorView>)
        const p = screen.getByLabelText('error')
        expect(p.classList.toString()).toContain('bg-red-800')
    });
});