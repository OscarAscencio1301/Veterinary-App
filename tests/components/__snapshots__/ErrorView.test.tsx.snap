// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component ErrorView Snapshot 1`] = `
<div>
  <div
    aria-label="error"
    class="bg-red-800 text-white rounded py-2"
  >
    <p>
      <p>
        Error Testing
      </p>
    </p>
  </div>
</div>
`;
